resolwe/flow/executors/startup_communication_container.py

```diff
@@ -75,6 +95,218 @@ def handle_error(func, path, exc_info):
+                )
+        return dict(zip(filenames, fds))
+
+    def run(self):
+        """Start listening for file descriptors.
+
+        Socket must be constructed here since it is not thread safe.
+        """
+        server_socket = socket.socket(family=socket.AF_UNIX)
+        server_socket.settimeout(PROCESSING_CONTAINER_TIMEOUT)
+        server_socket.bind(os.fspath(UPLOAD_FILE_SOCKET))
+        server_socket.listen()
+        logger.debug("Upload socket started: %s.", os.fspath(UPLOAD_FILE_SOCKET))
+        # Wait for the connection up to PROCESSING_CONTAINER_TIMEOUT seconds.
+        # If it fails, the socket will be closed and process terminated if if
+        # it will save some files as outputs.
+        try:
+            client, info = server_socket.accept()
+        except socket.timeout:
+            logger.error("Processing container is not connected to the upload socket")
+            self.upload_socket.close()
+            return
+
+        # Set the timeout for blocking operations to 1 second so we can
+        # check for terminating condition.
+        client.settimeout(1)
+        logger.info("Uploader client connected (%s)." % client)
+        with client:
+            while not self._terminating:
+                try:
+                    file_descriptors = self.receive_file_descriptors(client)
+                    # Stop if the socket has been closed.
+                    if not file_descriptors:
+                        break
+                except socket.timeout:
+                    logger.info("Uploader timeout waiting for data.")
+                    continue
+                except:
+                    logger.exception(
+                        "Exception while receiving file descriptors, exiting."
+                    )
+                    break
+                try:
+                    to_transfer = []
+                    logger.debug("Got %s", file_descriptors)
+                    referenced_files: dict[str, dict[str, str]] = dict()
+                    for file_name, file_descriptor in file_descriptors.items():
+                        file_size = os.stat(file_descriptor).st_size
+                        with os.fdopen(os.dup(file_descriptor), "rb") as stream:
+                            hasher = StreamHasher()
+                            hasher.compute(stream)
+                            referenced_files[file_name] = {
+                                hash_type: hasher.hexdigest(hash_type)
+                                for hash_type in StreamHasher.KNOWN_HASH_TYPES
+                            }
+                            referenced_files[file_name]["path"] = file_name
+                            referenced_files[file_name]["size"] = file_size
+
+                    logger.debug("Computed hashes: %s", referenced_files)
+                    # Copy the local objects to the disk using shutil.
+                    if UPLOAD_CONNECTOR_NAME == "local":
```

Comment:
The transfer method always add `url` between base path and file name.
So file `test.file` would be copied to `/data/url/test_file` instead of `/data/test_file`.

Reply:
I prefer to use a connector anyway. One option is to create a local connector that does not include the URL in the path, and the other solution would be to adjust the mount point to include the url.

Reply:
The URL in actually included by the transfer module.
And it is not so easy to fix. 

Mounting the volume differently could work, I will take a look into that.

Reply:
I believe my brains are half-fried. I looked at the code again and remerbered that the local connector actually points to the shared filesystem and there path must be added. So I just removed the local part and it "just works".